### PR TITLE
Add extended tests for pipeline examples

### PR DESCRIPTION
**Pull request by csrwng**
Moves common code out of Jenkins plugin test to utility package.
Adds tests to build pipeline to validate new pipeline examples.
Fixes bug in bluegreen pipeline.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

File: test/extended/builds/pipeline.go [hunk 1]
```diff
@@ -2,19 +2,27 @@ package builds
...
+	"github.com/openshift/origin/test/extended/util/jenkins"
 )
 
 var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 	defer g.GinkgoRecover()
 	var (
-		pipelineTemplatePath = exutil.FixturePath("testdata", "test-pipeline.json")
-		jenkinsTemplatePath  = exutil.FixturePath("..", "..", "examples", "jenkins", "jenkins-ephemeral-template.json")
-		oc                   = exutil.NewCLI("jenkins-pipeline", exutil.KubeConfigPath())
+		pipelineTemplatePath      = exutil.FixturePath("testdata", "test-pipeline.json")
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: do we actually need test-pipeline if we're testing other real pipelines?

**Comment by csrwng** [comment]
[testextended][extended:core(openshift pipeline build)]

File: test/extended/builds/pipeline.go [hunk 2]
```diff
@@ -50,4 +58,133 @@ var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
...
+	g.Context("Blue-green pipeline", func() {
+		g.It("Should build and complete successfully", func() {
+			// Deploy Jenkins without oauth
+			g.By(fmt.Sprintf("calling oc new-app -f %q -p ENABLE_OAUTH=false", jenkinsTemplatePath))
+			err := oc.Run("new-app").Args("-f", jenkinsTemplatePath, "-p", "ENABLE_OAUTH=false").Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Wait for Jenkins to deploy
+			g.By("waiting for Jenkins to deploy")
+			err = exutil.WaitForADeploymentToComplete(oc.KubeClient().Core().ReplicationControllers(oc.Namespace()), "jenkins", oc)
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: you don't wait for the jenkins deployment in the other tests, why do you need to wait for it here?

Comment (csrwng) [inline test/extended/builds/pipeline.go]: so I can build the ref below (line 111) ... but I guess it can be done without a running service.

Comment (bparees) [inline test/extended/builds/pipeline.go]: yeah my read of the code was that it didn't depend on the service being up, just existing.

Comment (csrwng) [inline test/extended/builds/pipeline.go]: should I remove the wait?

File: test/extended/builds/pipeline.go (hunk 1 above)
Comment (csrwng) [inline test/extended/builds/pipeline.go]: I will remove

**Comment by csrwng** [comment]
Removed simple test and the wait for the jenkins service in the bluegreen test

File: test/extended/builds/pipeline.go [hunk 3]
```diff
@@ -2,52 +2,154 @@ package builds
...
 	o "github.com/onsi/gomega"
 
 	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/jenkins"
 )
 
 var _ = g.Describe("[builds][Slow] openshift pipeline build", func() {
 	defer g.GinkgoRecover()
 	var (
-		pipelineTemplatePath = exutil.FixturePath("testdata", "test-pipeline.json")
```
Comment (bparees) [inline test/extended/builds/pipeline.go]: remove the fixture too, unless something else is using it?

Comment (csrwng) [inline test/extended/builds/pipeline.go]: done

**Comment by bparees** [comment]
lgtm

**Comment by csrwng** [comment]
Looks like the Jenkins template is not added by default to the openshift namespace in the extended test environment. Now instantiating Jenkins explicitly.

**Comment by csrwng** [comment]
[merge]

**Comment by csrwng** [comment]
Flake https://github.com/openshift/origin/issues/11772
[merge]

**Comment by csrwng** [comment]
Flake https://github.com/openshift/origin/issues/12251
[merge]

**Pull request merged**